DigitalOcean 1-click Droplet and Kubernetes getting started guides (#8773)
gravitational/teleport: docs/pages/getting-started/digitalocean.mdx

---
title: Getting started with Teleport on DigitalOcean
description: How to install Teleport on DigitalOcean?
videoBanner: voHQlSX_czE
---

This tutorial will guide you through quickly getting started with Teleport on DigitalOcean with Teleport 1-Click Droplet app.
 
<Admonition
  type="note"
  title="Note"
>
  This tutorial covers getting started with Teleport using DigitalOcean 1-Click droplet app. If you are looking for a manual installation, refer to our [Linux installation guide](./linux-server.mdx). 

</Admonition>

Prerequisites
- A Fully Qualified Domain Name (this will be used as a [clustername](../setup/admin/trustedclusters.mdx#introduction) for Teleport).
- A two-factor authenticator app(e.g., [Google Authenticator](https://www.google.com/landing/2step/)).
- An SSH client like OpenSSH.

> ## Step 1/4. Install Teleport with 1-Click

Reviewer: The hyphen between "1" and "Click" is how DigitalOcean [write on their website](https://www.digitalocean.com/products/marketplace/). So let's keep it as is in the original source?

> ## Step 1/4. Install Teleport with 1-Click

Reviewer 2: Sounds good!

Head over to Teleport page on [DigitalOcean marketplace](https://marketplace.digitalocean.com/apps/teleport) and click “Create a Droplet” button:

> ![Teleport 1-Click Droplet Page](../../img/quickstart/digitalocean/1click-droplet-page.png)

Reviewer: Yes, Teleport is already updated to v8.01 in the marketplace. I will be updating this guide with latest screenshot images.

> Once you click the button, DigitalOcean redirects you to the control panel where you can select VPS type, region, and set up SSH auth method. Teleport server is very lightweight, and if you are just trying out Teleport, you can select $5 droplet. Make sure you select SSH keys as the SSH authentication method as it is more secure than a password.

Reviewer: yes the VPS is for the whole Teleport deployment.

<Admonition
  type="tip"
  title="Tip"
>

>   Teleport supports certificate-based auth for SSH, which is more secure than both keys and passwords. 

Reviewer: Nope. Removing this one.

___
• Enter Teleport cluster name (FQDN): dodemo.teleporters.dev
• Enter your email address to retrieve TLS certificate from Lets Encrypt: dodemo@teleporters.dev
• Enter a username for the initial Teleport user: tadmin

Initializing...
[+] Generating new host UUID...
[+] Updating cluster networking configuration...
[+] Generating user and host certificate authority...
[+] Enabling RBAC in OSS Teleport. Migrating users, roles and trusted clusters...
[+] Starting Auth and Proxy services...
[+] Final Checks...

***************************************************************************
**                             ---
**    Teleport is configured and user tadmin has been created
**    but requires a password. Open the URL link below to complete the
**    setup. The link is valid for 1h:
**
**    https://dodemo.teleporters.dev:443/web/invite/992a29e8ef845d2f0189dee316f92939
**
**                             ---
**                     HAPPY TELEPORTING :)
***************************************************************************
```

Copy the URL link printed in this step:
`https://dodemo.teleporters.dev:443/web/invite/992a29e8ef845d2f0189dee316f92939`

> This link opens up Teleport Web UI, where you will need to set a password and configure two-factor authentication to complete the user setup process.

Reviewer: "Web UI" is how we refer to Teleport's web UI in website and docs.

## Step 3/4. Complete user setup and login to Teleport UI
Open the link copied in the previous step in the browser to complete the setup process. When the web page is ready:
1. Scan the QR code with two-factor authentication app (e.g., Google Authenticator)
2. Set a password and enter TOTP code generated from the two-factor Authentication app.
![Setup User](../../img/quickstart/digitalocean/setup-user.png)

Once you setup a password and provide a valid TOTP code, the user setup process will be complete, and you will be redirected to WebUI:

![Teleport WebUI](../../img/quickstart/digitalocean/webui.png)

Congrats! You've completed setting up Teleport! Finally, you are a step closer to managing secure access to your infrastructure hosted in DigitalOcean. 

## Step 4/4. Protect access to DigitalOcean Services with Teleport!

> The next step for you should be to protect direct SSH access to your droplets using Teleport. You can also secure access to Database, Marketplace apps, and Kubernetes Cluster using Teleport. Below are the links to get started further:

Reviewer: > I think the benefit of performing each next step should be more specific in relation to a user's DigitalOcean environment.

I agree. We need individual "how-to" guides specific to DigitalOcean. To begin with, even a simple "how to protect ssh access with Teleport" would suffice. Since we do not have those tutorials available, I thought the tiles should help user to navigate next.

Head over to the Teleport page on [DigitalOcean Marketplace](https://marketplace.digitalocean.com/apps/teleport) and click the “Create a Droplet” button:

<Figure align="left" bordered caption="Teleport 1-Click droplet page">
  ![Teleport 1-Click droplet page](../../img/quickstart/digitalocean/1click-droplet-page.png)
</Figure>

Once you click the button, DigitalOcean redirects you to the control panel to configure resources for the Teleport droplet. This step is similar to how you create a regular [droplet in DigitalOcean](https://docs.digitalocean.com/products/droplets/how-to/create/). Teleport is very lightweight, and if you are just trying out Teleport, you can select the $5 droplet. Make sure you select "SSH keys" as the SSH authentication method as it is more secure than a password.
<Figure align="left" bordered caption="Create a droplet">
  ![Create a droplet](../../img/quickstart/digitalocean/create-droplet.png)
</Figure>

It will take a few minutes before our newly created Teleport droplet is ready. Once the droplet is ready, configure your FQDN with the public IP address of the droplet as an IP address for the `A` record of your domain name. 

> For example, refer to the image below; we use the domain name `teleporters.dev`. The resulting domain we are using as an FQDN is `dodemo.teleporters.dev`, pointing to our Teleport droplet's public IP `192.168.200.200`.

Reviewer 3: ```suggestion
For example, refer to the image below; we use the domain name `example.com`. The resulting domain we are using as an FQDN is `tele.example.com`, pointing to our Teleport droplet's public IP `192.168.200.200`.
```
Updated to be consistent with https://goteleport.com/docs/getting-started/linux-server/#configure-dns

[+] Updating cluster networking configuration...
[+] Generating user and host certificate authority...
[+] Enabling RBAC in OSS Teleport. Migrating users, roles and trusted clusters...
[+] Starting Auth and Proxy services...
[+] Final Checks...

***************************************************************************
**                             ---
**    Teleport is configured and user tadmin has been created
**    but requires a password. Open the URL link below to complete the
**    setup. The link is valid for 1h:
**
**    https://dodemo.teleporters.dev:443/web/invite/<invite-token>
**
**                             ---
**                     HAPPY TELEPORTING :)
***************************************************************************
```

Copy the URL link printed in this step:
`https://dodemo.teleporters.dev:443/web/invite/<invite-token>`

This link opens up Teleport Web UI, where you will need to set a password and configure two-factor authentication to complete the user setup process.

## Step 3/3. Complete user setup and log in to Teleport UI
Open the link copied in the previous step in the browser to complete the setup process. When the web page is ready:

> 1. Scan the QR code with your two-factor authentication app (e.g., Google Authenticator)

Reviewer 3: If using Teleport 8+ we can use WebAuth and provide the option for a hardware or OTP for 2nd factor? Maybe an addition for the next version.

2. Set a password and enter the TOTP code generated from the two-factor authentication app.
<Figure align="left" bordered caption="Set up user">
  ![Set up user](../../img/quickstart/digitalocean/setup-user.png)
</Figure>

Once you set up a password and provide a valid TOTP code, the user setup process will be complete, and you will be redirected to Teleport Web UI:

<Figure align="left" bordered caption="Teleport Web UI">
  ![Teleport Web UI](../../img/quickstart/digitalocean/webui.png)
</Figure>

Congrats! You've completed setting up Teleport.

> ## Next steps.

Reviewer 2: ```suggestion
## Next steps
```

videoBanner: voHQlSX_czE
---

<Admonition
  type="note"
  title="Note"
>
This tutorial will guide you through quickly getting started with Teleport on DigitalOcean with Teleport 1-Click Droplet app. If you are looking for a manual installation, refer to our [Linux installation guide](./linux-server.mdx). 

</Admonition>

> **Prerequisites**

Reviewer 2: ```suggestion
## Prerequisites
```
So this appears in the Table of Contents

- A Fully Qualified Domain Name (FQDN).
- A two-factor authenticator app (e.g., [Google Authenticator](https://www.google.com/landing/2step/)).
- An SSH client like OpenSSH.

## Step 1/3. Install Teleport with 1-Click

> Head over to the Teleport page on [DigitalOcean marketplace](https://marketplace.digitalocean.com/apps/teleport) and click the “Create a Droplet” button:

Reviewer 2: ```suggestion
Head over to the Teleport page on [DigitalOcean Marketplace](https://marketplace.digitalocean.com/apps/teleport) and click the “Create a Droplet” button:
```

I think this is capitalized (https://www.digitalocean.com/blog/introducing-digitalocean-marketplace/).

**    but requires a password. Open the URL link below to complete the
**    setup. The link is valid for 1h:
**
**    https://dodemo.teleporters.dev:443/web/invite/<invite-token>
**
**                             ---
**                     HAPPY TELEPORTING :)
***************************************************************************
```

Copy the URL link printed in this step:
`https://dodemo.teleporters.dev:443/web/invite/<invite-token>`

This link opens up Teleport Web UI, where you will need to set a password and configure two-factor authentication to complete the user setup process.

> ## Step 3/3. Complete user setup and login to Teleport UI

Reviewer 2: ```suggestion
## Step 3/3. Complete user setup and log in to Teleport UI
```
